server/src/main/java/org/apache/iotdb/db/metadata/template/Template.java

```diff
@@ -113,12 +136,20 @@ public void setSchemaMap(Map<String, IMeasurementSchema> schemaMap) {
   /**
    * check whether a timeseries path is compatible with this template
    *
+   * <p>Adaptation for tree-structured template: Iff the last or penultimate phase of the path match
+   * the direct measurement of template, return false
+   *
    * @param path timeseries path
    * @return whether we can create this new timeseries (whether it's compatible with this template)
    */
+  @Deprecated
   public boolean isCompatible(PartialPath path) {
-    return !(schemaMap.containsKey(path.getMeasurement())
-        || schemaMap.containsKey(path.getDevicePath().getMeasurement()));
+    if (mountedNode.hasChild(path.getMeasurement())
+        && mountedNode.getChild(path.getMeasurement()).isMeasurement()) return false;
+    if (mountedNode.hasChild(path.getDevicePath().getMeasurement())
+        && mountedNode.getChild(path.getDevicePath().getMeasurement()).isMeasurement())
+      return false;
+    return true;
   }
```

Comment:
If no more usage, delete this method directly.

Reply:
it's removed and substituted with a new method:
public boolean isDirectNodeInTemplate(String nodeName);